pandas/core/dtypes/base.py

```diff
@@ -422,28 +427,52 @@ def register(self, dtype: type[ExtensionDtype]) -> None:
 
         self.dtypes.append(dtype)
 
-    def find(self, dtype: type[ExtensionDtype] | str) -> type[ExtensionDtype] | None:
+    @overload
+    def find(self, dtype: type[ExtensionDtypeT]) -> ExtensionDtypeT:
+        ...
+
+    @overload
+    def find(self, dtype: ExtensionDtypeT) -> ExtensionDtypeT:
+        ...
+
+    @overload
+    def find(
+        self, dtype: NpDtype | type_t[str | float | int | complex | bool | object] | str
```

Comment:
type_t and type used. choose one for consistency.

Reply:
changed to `type_t` in next commit